MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -292,45 +296,98 @@ void vtkSlicerMarkupsToModelLogic::UpdateOutputCloseSurfaceModel(vtkMRMLMarkupsT
   vtkSmartPointer< vtkCellArray > modelCellArray = vtkSmartPointer< vtkCellArray >::New();
 
   modelPoints->SetNumberOfPoints(numberOfMarkups);
-  double markupPoint [3] = {0.0, 0.0, 0.0};
+  modelCellArray->InsertNextCell(numberOfMarkups);
 
-  for (int i=0; i<numberOfMarkups;i++)
-  {
-    markups->GetNthFiducialPosition(i,markupPoint);
-    modelPoints->SetPoint(i, markupPoint);
-  }
+  double markupPoint[3] = {0.0, 0.0, 0.0};
+  double coords[numberOfMarkups][3];
+  double meanPoint[3] = {0.0, 0.0, 0.0};
 
-  modelCellArray->InsertNextCell(numberOfMarkups);
-  for (int i=0; i < numberOfMarkups;i++)
+  for (int i = 0; i < numberOfMarkups; i++)
   {
+    markups->GetNthFiducialPosition(i, markupPoint);
+    modelPoints->SetPoint(i, markupPoint);
+    vtkMath::Add(meanPoint, markupPoint, meanPoint);
+    coords[i][0] = markupPoint[0];
+    coords[i][1] = markupPoint[1];
+    coords[i][2] = markupPoint[2];
     modelCellArray->InsertCellPoint(i);
   }
+  vtkMath::MultiplyScalar(meanPoint, 1.0 / numberOfMarkups);
+
```

Comment:
I don't understand easily what you are computing and why. Could you please comment it better? Maybe also move the computation into a separate method.